src/main/flatbuffers/rlbot.fbs

```diff
@@ -292,171 +292,418 @@ table DesiredGameState {
-    team: int;
-    boost: int;
+  location: Vector3;
+  rotation: Rotator;
+  velocity: Vector3;
+  hasWheelContact: bool;
+  isSupersonic: bool;
+  team: int;
+  boost: int;
 }
 
 /// A minimal version of the ball, useful when bandwidth needs to be conserved.
 table TinyBall {
-    location: Vector3;
-    velocity: Vector3;
+  location: Vector3;
+  velocity: Vector3;
 }
 
 /// A minimal version of the game tick packet, useful when bandwidth needs to be conserved.
 table TinyPacket {
-    players: [TinyPlayer];
-    ball: TinyBall;
+  players: [TinyPlayer];
+  ball: TinyBall;
 }
 
 table PredictionSlice {
-    /// The moment in game time that this prediction corresponds to.
-    /// This corresponds to 'secondsElapsed' in the GameInfo table.
-    gameSeconds: float;
+  /// The moment in game time that this prediction corresponds to.
+  /// This corresponds to 'secondsElapsed' in the GameInfo table.
+  gameSeconds: float;
 
-    /// The predicted location and motion of the object.
-    physics: Physics;
+  /// The predicted location and motion of the object.
+  physics: Physics;
 }
 
 table BallPrediction {
-    /// A list of places the ball will be at specific times in the future.
-    /// It is guaranteed to sorted so that time increases with each slice.
-    /// It is NOT guaranteed to have a consistent amount of time between slices.
-    slices: [PredictionSlice];
+  /// A list of places the ball will be at specific times in the future.
+  /// It is guaranteed to sorted so that time increases with each slice.
+  /// It is NOT guaranteed to have a consistent amount of time between slices.
+  slices: [PredictionSlice];
+}
+
+/// A bot controlled by the RLBot framework
+table RLBotPlayer {}
+
+/// A normal human player
+table HumanPlayer {}
+
+/// A psyonix bot, e.g. All Star bot
+table PsyonixBotPlayer {
+  botSkill: float;
+}
+
+/// A player that Rocket League treats as human, e.g. has a dedicated camera and can do training mode,
+/// but is actually controlled by a bot.
+table PartyMemberBotPlayer {}
+
+union PlayerClass { RLBotPlayer, HumanPlayer, PsyonixBotPlayer, PartyMemberBotPlayer }
+
+/// The car type, color, and other aspects of the player's appearance.
+/// See https://github.com/RLBot/RLBot/wiki/Bot-Customization
+table PlayerLoadout {
+  teamColorId: int;
+  customColorId: int;
+  carId: int;
+  decalId: int;
+  wheelsId: int;
+  boostId: int;
+  antennaId: int;
+  hatId: int;
+  paintFinishId: int;
+  customFinishId: int;
+  engineAudioId: int;
+  trailsId: int;
+  goalExplosionId: int;
+  loadoutPaint: LoadoutPaint;
+}
+
+/// Specification for 'painted' items. See https://github.com/RLBot/RLBot/wiki/Bot-Customization
+table LoadoutPaint {
+  carPaintId: int;
+  decalPaintId: int;
+  wheelsPaintId: int;
+  boostPaintId: int;
+  antennaPaintId: int;
+  hatPaintId: int;
+  trailsPaintId: int;
+  goalExplosionPaintId: int;
+}
+
+table PlayerConfiguration {
+  playerClass: PlayerClass;
+  name: string;
+  team: ubyte;
```

Comment:
This is inconsistent with TinyPlayer which stores this as an int.
How about storing it as an enum in both cases?

Reply:
Good call. It's also stored as an int in PlayerInfo, so I'll be consistent with that. I'm not keen on switching to an enum in those existing tables, because we'd have to leave the int around as a deprecated field to avoid breaking backwards compatibility. Plus all the bots use int for team and they're not going to migrate any time soon.